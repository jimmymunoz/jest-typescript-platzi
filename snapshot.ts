export const getCharacter = ({id, name, status, species, gender}) => ({
  id,
  name,
  status,
  species,
  gender
})