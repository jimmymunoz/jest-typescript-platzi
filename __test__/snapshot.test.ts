import { getCharacter } from '../snapshot';
import rick from '../rick.json'

// npx jest __test__/snapshot.test.ts
describe('Es hora de las instantaneas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  });

  test.skip('Siempre fallarara el snapshot', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() *20),
    }
    expect(user).toMatchSnapshot()

  });

  test.only('Tenemos una exepcion en el condigo', () => {
    const user = {
      name: "Jimmy MUNOZ",
      id: Math.floor(Math.random() *20),
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    })

  });
});