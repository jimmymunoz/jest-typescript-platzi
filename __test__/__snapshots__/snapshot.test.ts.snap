// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallarara el snapshot 1`] = `
Object {
  "createAt": 2021-05-09T06:59:06.510Z,
  "id": 6,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una exepcion en el condigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Jimmy MUNOZ",
}
`;
